docs/docs/advanced-use/transient-traits.md

```diff
@@ -0,0 +1,587 @@
+```
+
+Mark the whole Identity as transient:
+
+```ruby
+$flags = $flagsmith.get_identity_flags('my-user', true, my_trait_key: 123.5, my_other_key: true)
+```
+
+</TabItem>
+<TabItem value="php" label="PHP">
+
+:::info
+
+Transient Traits and Identities are supported starting from PHP SDK version 4.4.0.
+
+:::
+
+Mark a Trait as transient:
+
+```php
+$flags = $flagsmith->getIdentityFlags('my-user', (object)['my_trait_key' => (object)['transient' => true, 'value' => 123.5], 'my_other_key' => true]);
+```
+
+Mark the whole Identity as transient:
+
+```php
+$flags = $flagsmith->getIdentityFlags('my-user', (object)['my_trait_key' => 123.5, 'my_other_key' => true], true);
+```
+
+</TabItem>
+<TabItem value="go" label="Go">
+
+:::info
+
+Transient Traits and Identities are supported starting from Go SDK version 4.0.0.
+
+:::
+
+Mark a Trait as transient:
+
+```go
+flags, _ := client.GetFlags(
+    ctx,
+    &flagsmith.NewEvaluationContext(
+        "my-user",
+        map[string]*interface{}{
+            "my_trait_key": NewTraitEvaluationContext(123.5, true),
+            "my_other_key": true,
+        },
+    )
+)
+```
+
+Mark the whole Identity as transient:
+
+```go
+flags, _ := client.GetFlags(
+    ctx,
+    &flagsmith.NewTransientEvaluationContext(
+        "my-user",
+        map[string]*interface{}{
+            "my_trait_key": 123.5,
+            "my_other_key": true,
+        },
+    )
+)
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+:::info
+
+Transient Traits and Identities are supported starting from Rust SDK version 2.0.0.
+
+:::
+
+Mark a Trait as transient:
+
+```rust
+use flagsmith::models::SDKTrait;
+use flagsmith_flag_engine::types::{FlagsmithValue, FlagsmithValueType};
+
+let identifier = "delboy@trotterstraders.co.uk";
+
+let traits = vec![
+    SDKTrait::new_with_transient(
+        "my_trait_key".to_string(),
+        FlagsmithValue {
+            value: "123.5".to_string(),
+            value_type: FlagsmithValueType::Float,
+        },
+        true,
+    ),
+    SDKTrait::new(
+        "my_other_key".to_string(),
+        FlagsmithValue {
+            value: "true".to_string(),
+            value_type: FlagsmithValueType::Bool,
+        },
+        true,
+    ),
+];
+
+let identity_flags = flagsmith.get_identity_flags(identifier, Some(traits), None).unwrap();
+```
+
+Mark the whole Identity as transient:
+
+```rust
+use flagsmith::models::SDKTrait;
+use flagsmith_flag_engine::types::{FlagsmithValue, FlagsmithValueType};
+
+let identifier = "delboy@trotterstraders.co.uk";
+
+let traits = vec![
+    SDKTrait::new(
+        "my_trait_key".to_string(),
+        FlagsmithValue {
+            value: "123.5".to_string(),
+            value_type: FlagsmithValueType::Float,
+        },
+    ),
+    SDKTrait::new(
+        "my_other_key".to_string(),
+        FlagsmithValue {
+            value: "true".to_string(),
+            value_type: FlagsmithValueType::Bool,
+        },
+        true,
```

Comment:
Maybe I'm wrong above - what is this argument for?

Reply:
This was incorrect; fixed.